discord/member.go

```diff
@@ -83,7 +83,7 @@ type MemberUpdate struct {
 	CommunicationDisabledUntil *json.Nullable[time.Time] `json:"communication_disabled_until,omitempty"`
 }
 
-// SelfNickUpdate is used to update your own nick
-type SelfNickUpdate struct {
-	Nick string `json:"nick"`
+// CurrentMemberUpdate is used to update the current member
+type CurrentMemberUpdate struct {
+	Nick *string `json:"nick"`
```

Comment:
empty string should reset this field

Reply:
passing `nil` is definitely better UX.

Reply:
creating a pointer is simply annoying
I'd prefer to avoid them where possible